src/AppBundle/Controller/BlogController.php

```diff
@@ -118,4 +119,48 @@ public function commentFormAction(Post $post)
             'form' => $form->createView(),
         ));
     }
+
+    /**
+     * @Route("/search", name="blog_search")
+     * @Method("GET")
+     *
+     * @return JsonResponse
+     */
+    public function searchAction(Request $request)
+    {
+        $query = $request->query->get('q', '');
+
+        // Sanitizing the query: removes all non-alphanumeric characters except whitespaces
+        $query = preg_replace('/[^[:alnum:] ]/', '', trim(preg_replace('/[[:space:]]+/', ' ', $query)));
+
+        // Splits the query into terms and removes all terms which
+        // length is less than 2
+        $terms = array_unique(explode(' ', strtolower($query)));
+        $terms = array_filter($terms, function($term) {
+            return 2 <= strlen($term);
+        });
+
+        $posts = array();
+
+        if (!empty($terms)) {
+            $posts = $this
+                ->getDoctrine()
+                ->getManager()
+                ->getRepository('AppBundle:Post')
+                ->findByTerms($terms)
+            ;
+        }
+
+        $results = array();
+
+        foreach ($posts as $post) {
+            array_push($results, array(
+                // We should never trust to admin user
+                'result' => strip_tags($post->getTitle()),
```

Comment:
if your contract is that ``results`` must be an HTML string, this is the wrong way to produce this HTML. you are not returning an HTML string here.